hal/src/nRF52840/ble_hal.cpp

```diff
@@ -979,7 +981,9 @@ int BleObject::Broadcaster::setAdvertisingData(const uint8_t* buf, size_t len) {
         len = 0;
     }
     advDataLen_ = len;
-    configure(nullptr);
+    if (configure(nullptr) != SYSTEM_ERROR_NONE) {
```

Comment:
Should this error be propagated?

Reply:
Same in https://github.com/particle-iot/device-os/pull/1891/files#diff-fba7ee4853a5cf25cf4e9837d8c5f803R1009